app/services/idv/session.rb

```diff
@@ -99,6 +99,11 @@ def vendor_params
       applicant.merge('uuid' => current_user.uuid)
     end
 
+    def profile_id=(value)
```

Comment:
This was necessary because `confirm_no_profile_yet` introduces a new read of `idv_session.profile` early on in the request, which can lead to `nil` getting cached as `idv_session.profile`. This just makes sure that we bust the cache when updating `profile_id`